readme/MacBuild.md

```diff
@@ -1,34 +1,175 @@
+https://brew.sh/
+
+````sh
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+````
+
+### Install dependencies
+````sh
+brew install cmake ninja pkg-config gtk+3 poppler librsvg adwaita-icon-theme libzip portaudio libsndfile
+````
 
-### 1. Build GTK
-````bash
-./build-gtk3.sh
+### Build Xournal++:
+````sh
+git clone http://github.com/xournalpp/xournalpp.git
+cd xournalpp
+mkdir build
+cd build
+cmake .. -GNinja
+cmake --build .
 ````
-#### Potential Errors
-##### itstool (version 2.0.6 on macOS High Sierra 10.13)
+
+## Bundling Xournal++ into a `.app` with gtk-osx
+
+This uses the official [gtk-osx][gtk-osx] project, which builds GTK using the
+jhbuild tool and a curated set of dependencies that allows GTK to work on macOS.
+
+For the easiest build experience, do not have MacPorts or Homebrew installed.
+* If you have any of them installed, you need to create a new user, and use the
+  new user for the whole process. This is because the jhbuild packages are meant
+  to be self-contained, and environment variables from MacPorts or Homebrew
+  configurations may interfere with the jhbuild package builds.
+* Even if you create a new user, packages built by jhbuild may still "find"
+  MacPorts or Homebrew packages. If this is the case, you can either uninstall
+  the corresponding packages; or (for advanced users) you can clone gtk-osx
+  and manually create a jhbuild _moduleset_ that explicitly ensures that such
+  packages are disabled (see the FAQ).
+* One possible way to use jhbuild alongside brew is to unlink all brew modules
+  before running jhbuild. After finishing the build, they can be relinked.
+  Note: this is untested though, and might fail.
+
+### A note on build dependencies
+
+All of the following steps, except for the last one, only need to be done once.
+
+Please take the OS version dependent problems for each step into account.
+The errors will vary with different versions of the libraries.
+If you encounter a yet unknown error, feel free to submit a pull request to add it to this set of instructions.
+
+
+### Step 1: Install jhbuild
+
+To install jhbuild, run the following commands. **Do not blindly copy and paste.**
+Make sure you understand what each command does as you run them one-by-one.
+
+```sh
+# Download the setup script
+curl -LR https://gitlab.gnome.org/GNOME/gtk-osx/raw/master/gtk-osx-setup.sh -o gtk-osx-setup.sh
+
+# Override Python version in setup script
+export PYTHON_VERSION=3.9.15
+
+# If jhbuild is already installed with a different python version, you MUST delete it first.
+rm -rf ~/gtk ~/.new_local ~/.config/jhbuildrc ~/.config/jhbuildrc-custom
+
+# workaround for openssl 1.1.1q compilation error:
+# https://github.com/openssl/openssl/issues/18720
+export OPENSSL_CFLAGS='-Wno-error=implicit-function-declaration'
+
+# Build jhbuild
+bash gtk-osx-setup.sh
+
+# add jhbuild to path
+echo 'export PATH="$HOME"/.new_local/bin:"$PATH"' >> ~/.zshrc
```

Comment:
After this, the console should be reopened, since otherwise the old `PATH` is still in use.

Reply:
The user could also run that export command directly. I've added methods below this line.